packages/fx-core/src/core/middleware/MigrationUtils.ts

```diff
@@ -210,7 +210,9 @@ export function namingConverterV3(
         case FileType.CONFIG:
           return ok(`${configPrefix}${res}`);
         case FileType.USERDATA:
-          return ok(`${secretPrefix}${res}`);
+          if (res.includes("STATE__"))
```

Comment:
it might be better to use:
  _if (res.startswith("STATE__"))_ 
in case that the env name includes "STATE__" in the middle of it
@blackchoey what do u think